l10n_ch_invoice_reports/data/reports.xml

```diff
@@ -8,7 +8,7 @@
         name="l10n_ch_invoice_reports.account_move_payment_report"
         file="l10n_ch_invoice_reports.account_move_payment_report"
         attachment_use="True"
-        attachment="invoice_'+(object.name)+'_with_payslip.pdf"
```

Comment:
object.name wasn't correctly replaced by invoice name on mail.template attachment, decided to update  it in model.